### PR TITLE
Feat grading

### PR DESCRIPTION
**Pull request by machow**
* adds ipython shell to bash kernel
* cells beginning with `#DC_GRADING` execute python
* ipython_nose ext loaded by default
* can use `%%nose` magic commands

see https://github.com/datacamp/ipython_nose/pull/4

### CODE REVIEW

File: bash_kernel/install.py [hunk 1]
```diff
@@ -21,7 +21,11 @@ def install_my_kernel_spec(user=True, prefix=None):
         # TODO: Copy resources once they're specified
 
         print('Installing IPython kernel spec')
+        # TODO: remove line below once projects backend is updated w/ bash config
+        KernelSpecManager().install_kernel_spec(td, 'python3', user=user, replace=True, prefix=prefix)
```
Comment (rv2e) [inline bash_kernel/install.py:25]: Can we remove it now?

File: bash_kernel/kernel.py [hunk 2]
```diff
@@ -129,6 +135,53 @@ def process_output(self, output):
 
     def do_execute(self, code, silent, store_history=True,
                    user_expressions=None, allow_stdin=False):
+
+        if code.startswith('#DC_GRADE'):
```
Comment (rv2e) [inline bash_kernel/kernel.py:139]: Can't you introspect the metadata of the cell in order to know if it's a test cell?

**Review by rv2e — COMMENTED** [review]
Just have two questions :)

File: bash_kernel/kernel.py (hunk 2 above)
Comment (machow) [reply bash_kernel/kernel.py:139]: I'm not sure. It seems like it should be possible, but I don't know how to simply introspect the full cell data :/

Comment (machow) [reply bash_kernel/kernel.py:139]: If we slot it in later it will be backwards compatible though ;)

**Pull request merged**